SerialPrograms/Source/PokemonBDSP/Programs/Farming/PokemonBDSP_PoffinCooker.cpp

```diff
@@ -0,0 +1,211 @@
+/*  Poffin Cooker
+ *
+ *  From: https://github.com/PokemonAutomation/Arduino-Source
+ *
+ */
+
+#include "CommonFramework/Tools/StatsTracking.h"
+#include "CommonFramework/ImageTools/ImageStats.h"
+#include "CommonFramework/ImageTools/ImageBoxes.h"
+#include "CommonFramework/VideoPipeline/VideoFeed.h"
+#include "CommonFramework/Notifications/ProgramNotifications.h"
+#include "NintendoSwitch/Commands/NintendoSwitch_Commands_PushButtons.h"
+#include "Pokemon/Pokemon_Strings.h"
+#include "PokemonBDSP/PokemonBDSP_Settings.h"
+#include "PokemonBDSP_PoffinCooker.h"
+
+namespace PokemonAutomation{
+namespace NintendoSwitch{
+namespace PokemonBDSP{
+    using namespace Pokemon;
+
+
+PoffinCooker_Descriptor::PoffinCooker_Descriptor()
+    : SingleSwitchProgramDescriptor(
+        "PokemonBDSP:PoffinCooker",
+        STRING_POKEMON + " BDSP", "Poffin Cooker",
+        "ComputerControl/blob/master/Wiki/Programs/PokemonBDSP/PoffinCooker.md",
+        "Cook Poffins.",
+        FeedbackType::REQUIRED,
+        AllowCommandsWhenRunning::DISABLE_COMMANDS,
+        PABotBaseLevel::PABOTBASE_12KB
+    )
+{}
+struct PoffinCooker_Descriptor::Stats : public StatsTracker{
+    Stats()
+        : m_attempts(m_stats["Poffins cooked"])
+    {
+        m_display_order.emplace_back("Poffins cooked");
+    }
+    std::atomic<uint64_t>& m_attempts;
+};
+std::unique_ptr<StatsTracker> PoffinCooker_Descriptor::make_stats() const{
+    return std::unique_ptr<StatsTracker>(new Stats());
+}
+
+
+PoffinCooker::PoffinCooker()
+    : GO_HOME_WHEN_DONE(false)
+    , MAX_COOK_ATTEMPTS(
+        "<b>Cook this many times:</b><br>This puts a limit on how many poffins you get. Don't forget that each cooking session gets you 4 poffins, and your bag cannot have more than 100 poffins. Thus you should never input more than 25 here.",
+        LockWhileRunning::LOCKED,
+        5
```

Comment:
You can have 25 as max in code to make sure users don't try dumb things

Reply:
Added to the code